opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -237,16 +251,13 @@ public void testInsertTaskWithStartAndDue()
 
         ), resultsIn(mClient,
                 new Assert<>(task, new TimeData(start, due)),
-                new AssertRelated<>(new InstanceTable(mAuthority), Instances.TASK_ID, task, new AllOf(
-                        new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
-                        new EqArg(Instances.INSTANCE_ORIGINAL_TIME, 0),
-                        new EqArg(Instances.INSTANCE_DUE, due.getTimestamp()),
-                        new EqArg(Instances.INSTANCE_START_SORTING, start.shiftTimeZone(TimeZone.getDefault()).getInstance()),
-                        new EqArg(Instances.INSTANCE_DUE_SORTING, due.shiftTimeZone(TimeZone.getDefault()).getInstance()),
-                        new EqArg(Instances.INSTANCE_DURATION, due.getTimestamp() - start.getTimestamp()),
-                        new EqArg(Tasks.TZ, start.isAllDay() ? "UTC" : start.getTimeZone().getID()),
-                        new EqArg(Instances.INSTANCE_STATUS, Instances.INSTANCE_STATUS_NEXT)
-                ))
+                new AssertRelated<>(
+                        new InstanceTable(mAuthority), Instances.TASK_ID, task,
+                        new InstanceData(
```

Comment:
Previous version checked `TZ` as well here.

By the way, there is something basic I don't understand here. Since `Instances` table has all the `TaskColumns` as well, does it mean that all those values are always in sync?
Shouldn't we always test both tables for those values then?

Reply:
No, we actually only need to test the task values in one of both tables. We just have to check the `TASK_ID` in the instances tables. Since we join both tables on this column we can be sure that the task values are correct in the instances view as well (if the `TASK_ID` is correct`). We should just add a test to check if the `join` is correct (i.e. contains all the columns).

In other words we don't have to test that the table join works, we just have to test that it's declared properly.

Reply:
Ok, I get it now that we use a View for Instances table which joins on Tasks, so they have be the same.
Worth creating a ticket now for testing the join, or not?

Reply:
Yeah, I guess a ticket would be good, although I'd consider this very low priority, because this is essentially covered by most provider tests.